hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/protocol/DatanodeDetails.java

```diff
@@ -504,20 +513,21 @@ public HddsProtos.DatanodeDetailsProto toProto(int clientVersion, Set<Port.Name>
    *                        If empty, all available ports will be included.
    * @return A {@link HddsProtos.DatanodeDetailsProto.Builder} Object.
    */
-
+  @VisibleForTesting
   public HddsProtos.DatanodeDetailsProto.Builder toProtoBuilder(
       int clientVersion, Set<Port.Name> filterPorts) {
 
-    HddsProtos.UUID uuid128 = HddsProtos.UUID.newBuilder()
-        .setMostSigBits(uuid.getMostSignificantBits())
-        .setLeastSigBits(uuid.getLeastSignificantBits())
-        .build();
+
+    HddsProtos.UUID uuid128 = id.toProto().getUuid();
 
     HddsProtos.DatanodeDetailsProto.Builder builder =
-        HddsProtos.DatanodeDetailsProto.newBuilder()
-            .setUuid128(uuid128);
+        HddsProtos.DatanodeDetailsProto.newBuilder();
 
-    builder.setUuidBytes(uuidString.getBytes());
+    builder.setId(id.toProto());
```

Comment:
`id.toProto()` does not need to be called twice.

Reply:
Fixed.